presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -392,10 +393,11 @@ public static Slice stBoundary(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     @SqlType(GEOMETRY_TYPE_NAME)
     public static Slice stEnvelope(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     {
-        OGCGeometry geometry = deserialize(input);
-        SpatialReference reference = geometry.getEsriSpatialReference();
-        Envelope envelope = getEnvelope(geometry);
-        return serialize(createFromEsriGeometry(envelope, reference));
+        Envelope envelope = deserializeEnvelope(input);
+        if (envelope == null) {
+            return EMPTY_POLYGON;
+        }
+        return serialize(createFromEsriGeometry(envelope, null));
```

Comment:
Previously, you were passing the spatial reference of `geometry` to `createFromEsriGeometry()`, but now you pass `null`. Why is that change needed?

Reply:
I no longer have the spatial reference of `geometry`, hence, cannot pass it. However, we don't use spatial reference anywhere and all geometries have `null` spatial reference because of `geometry.setSpatialReference(null);` in `geometryFromText`.